File: rest/java/storage/model/src/main/java/m4gshm/benchmark/rest/java/storage/ReactorStorage.java
package m4gshm.benchmark.rest.java.storage;

import m4gshm.benchmark.rest.java.storage.model.IdAware;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

import java.util.List;

public interface ReactorStorage<T extends IdAware<ID>, ID> {
    Mono<T> get(ID id);

    Flux<T> getAll();

    Mono<T> store(T entity);

    Mono<Boolean> delete(ID id);
}

File: rest/kotlin/storage/src/main/kotlin/m4gshm/benchmark/storage/ReactorStorage.kt
package m4gshm.benchmark.storage

import reactor.core.publisher.Mono

interface ReactorStorage<T, ID> {
    fun get(id: ID): Mono<T>
    fun store(id: ID, entity: T): Mono<Void>
    fun getAll(): Mono<List<T>>
    fun delete(id: ID): Mono<Boolean>
}